### PR TITLE
[apex] Add OneDeclarationPerStatement rule

### PR DESCRIPTION
**Pull request by jeffhube**
Please, prefix the PR title with the language it applies to within brackets, such as *[java]* or *[apex]*. If not specific to a language, you can use *[core]*

Before submitting a PR, please check that:
 - [x] The PR is submitted against `master`. The PMD team will merge back to support branches as needed.
 - [x] `./mvnw test` passes.
 - [x] `./mvnw pmd:check` passes.
 - [x] `./mvnw checkstyle:check` passes. [Check this for more info](https://github.com/pmd/pmd/blob/master/CONTRIBUTING.md#code-style)

**PR Description:**

This PR adds the OneDeclarationPerStatement rule to apex. It applies to field declarations (inside classes) and variable declarations (inside methods).

```
Integer a, b;   // not recommended

Integer a;      // preferred approach
Integer b;
```

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No java rules are changed!</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-apex/src/main/resources/category/apex/codestyle.xml [hunk 1]
```diff
@@ -149,6 +149,35 @@ public class Foo {
...
+        <description>
+Apex allows the use of several variables declaration of the same type on one line. However, it
+can lead to quite messy code. This rule looks for several declarations on the same line.
+        </description>
+        <priority>1</priority>
+        <properties>
+            <property name="xpath">
+                <value>
+<![CDATA[
+//VariableDeclarationStatements[VariableDeclaration[2]]|//FieldDeclarationStatements[FieldDeclaration[2]]
```
Comment (oowekyala) [inline pmd-apex/src/main/resources/category/apex/codestyle.xml]: I'd find it more readable to use the count function, e.g. `FieldDeclarationStatements[count(FieldDeclaration) > 1]`, and to  put the two alternatives on separate lines

File: pmd-apex/src/main/resources/category/apex/codestyle.xml [hunk 2]
```diff
@@ -149,6 +149,35 @@ public class Foo {
         </example>
     </rule>
 
+    <rule name="OneDeclarationPerStatement"
+          since="6.7.0"
+          message="Use one line for each declaration, it enhances code readability."
```
Comment (oowekyala) [inline pmd-apex/src/main/resources/category/apex/codestyle.xml]: Maybe "use one *statement* for each declaration"?

**Comment by jeffhube** [comment]
Fixed, should I include the changes to the autogenerated docs in this PR?

**Comment by jsotuyod** [comment]
@jeffhube no need to! The CI build will do so and push back to the repo. Thanks for the PR!

**Comment by jsotuyod** [comment]
For consistency, we may want to rename this and tune the rule to be equivalent to Java's [`OneDeclarationPerLine`](https://pmd.github.io/pmd-6.6.0/pmd_rules_java_bestpractices.html#onedeclarationperline)

**Pull request merged**